store/src/java/com/zimbra/cs/db/DbSearchHistory.java

```diff
@@ -197,11 +198,14 @@ public boolean isRegistered(DbConnection conn, String searchString) throws Servi
                     sb.append(DbUtil.whereIn("id", params.getIds().size()));
                 }
             } else {
-                sb.append(DebugConfig.disableMailboxGroups ? "" : "mailbox_id = ?");
+                sb.append(DebugConfig.disableMailboxGroups ? "" : "mailbox_id = ? ");
             }
-            sb.append(" ORDER BY last_search_date DESC ");
-            if (params.getNumResults() > 0 && Db.supports(Capability.LIMIT_CLAUSE)) {
-                sb.append(Db.getInstance().limit(params.getNumResults()));
+            if (!params.hasIds()) {
+                //no need to sort/limit otherwise, since results will be re-sorted according to relevance
```

Comment:
I don't understand what's happening here. If the results will be reordered according to relevance and the comment is saying that there is no need to sort, why add "ORDER BY date"?

Reply:
that might be a case of unclear commenting - it says "no need to sort/limit _otherwise_", referring to the case when the params _do_ have IDs. If we are in this _if_ block, then we did not have to go the index, meaning we need to sort chronologically.